src/main/java/com/google/api/gax/core/RetrySettings.java

```diff
@@ -151,6 +161,13 @@ public Builder toBuilder() {
     public abstract Builder setMaxRetryDelay(Duration maxDelay);
 
     /**
+     * MaxAttempts defines the maximum number of attempts to perform. If number of attempts reaches
+     * this limit the logic will give up retrying even if the total retry time is still lower than
+     * TotalTimeout.
+     */
+    public abstract Builder setMaxAttempts(int maxAttempts);
```

Comment:
There is a minimum attempts field in the gcloud RetryParam class. Are we deprecating that functionality?